linode/manager
# upcoming: [M3 7738] - Update Placement Group Create & Edit Drawers

## Description 📝
This brings a handful of updates to the Placement Group create/edit flow (functionality, nomenclature, cleanup). 

The UX/UI has warranted this change. We previously had a shared drawer for create & edit flows, which made sense since we were showing the same fields on both (some being disabled for the edit flow). Those requirements have now changed,  and the flows differences warrants two separate components. It helps with logic, readability and types.

There's few functionality changes, which I made sure to include in the testing steps and in the list below.

The diff is a bit larger than expected due to the naming convention changes.

## Changes  🔄
- Change "rename" in favor of "update" (code) or "edit" (copy)
- Split concern/.code between Create and Edit drawers
- Make Affinity Type select its own component for clarity/code consistency
- Default Affinity Type select to "Anti-affinity" and disable the "Affinity" option
  - "Anti-affinity" it will be the only option for phase 1 (new requirement)
  - We are keeping the disabled "Affinity" menu item(with a tooltip) as it will be a quick follow up and we want to expose users to the upcoming availability without altering the UX/UI
- Implement new Affinity Enforcement radio group
- Improve error handling

**Note**: we are aware of the links within the tooltips. I am leaving those for now until a better solution is provided, or we accept the lack of keyboard accessibility for phase 1 (TBD)

## Preview 📷
| Create Drawer  | Edit Drawer   |
| ------- | ------- |
|  ![localhost_3000_placement-groups_edit_5 (1)](https://github.com/linode/manager/assets/130582365/eed66e86-d0bd-4ed8-819d-905ac9d5b1ea) | ![localhost_3000_placement-groups_edit_1 (2)](https://github.com/linode/manager/assets/130582365/a1aa5f35-ac75-40ec-98cb-558f8161ceca) |

## How to test 🧪

### Prerequisites
- Pull code locally, turn on both the "Placement Group" feature flag and MSW

### Verification steps
- Navigate to http://localhost:3000/placement-groups
  - verify the create drawer by clicking on "Create Placement Group"
  - verify the create drawer by clicking on "Edit" for any of the placement groups

Things to verify:
- form validation
- error handling
- payload

Variation on the Create Drawer (with preselected region ID:
- Add a `selectedRegionId` to the [PlacementGroupLanding create drawer instance](https://github.com/linode/manager/blob/877e20604edc70f8ee3256012ebb5647218e5720/packages/manager/src/features/PlacementGroups/PlacementGroupsLanding/PlacementGroupsLanding.tsx#L195)
- Verify the create drawer by clicking on "Create Placement Group" (see below)

![localhost_3000_placement-groups_edit_1 (1)](https://github.com/linode/manager/assets/130582365/89a7b2b2-cc68-491f-b2f5-366f424d8c49)


⚠️ Copy is still very much in progress and will be updated in subsequent PRs (no need to check this PR for it)

## As an Author I have considered 🤔

*Check all that apply*
